core/src/smartcontracts/isi/world.rs

```diff
@@ -114,6 +114,30 @@ pub mod isi {
         }
     }
 
+    impl Execute for Register<PermissionTokenId> {
```

Comment:
Hmmm.... By design all permission tokens with this *id* should have the same parameters keys... Maybe it makes sense to pass expected parameters on `Register` step and then check if they exist on `Grant` step?

It's like dynamic typing

Reply:
It would make sense to add some schema-like thing to PermissionTokenDefinition, not sure if it's in scope for this PR thought.

Reply:
> It's like dynamic typing

It is the lisp variety of dynamic typing, to be more precise. 

> It would make sense to add some schema-like thing to PermissionTokenDefinition, not sure if it's in scope for this PR thought.

Can we generate the aforementioned schema? To me it looks like we shouldn't need to hard-code anything and instead move permission validators which depend on these parameters into run-time dependencies.

Reply:
@QuentinI , if you want this to be in the separate PR, please, create an issue for it. Seems like we decided that it worths implementation